### PR TITLE
Improve animation of new song select footer buttons

### PR DESCRIPTION
**Pull request by frenzibyte**
- Part of #22204 
- [x] Depends on #28141 (since `FooterButtonV2` is touched in there)

Changed to match animations of `ShearedButton`. Not a good idea to inherit from it since the footer button designs differ from `ShearedButton` slightly + we don't want the scale-in effect in footer buttons.

~~I have also fixed the "options" button not actually hiding when pressing on it while in open state. I'm not entirely sure about my approach but, well...it works™.~~

Preview:

https://github.com/ppy/osu/assets/22781491/f302043f-aefa-479e-994d-b54ab5dc8ccc

### CODE REVIEW

**Comment by Joehuu** [comment]
> I have also fixed the "options" button not actually hiding when pressing on it while in open state.

When I was implementing that, I mostly was following other popovers triggered by a button (e.g. add preset button). Pressing the button triggers the `DismissOnMouseDownContainer` on `PopoverContainer` then the button action which opens the popover again. I've found that behavior weird and expected toggling like done locally with options button.

**Comment by frenzibyte** [comment]
Well, yeah, that's what I'm achieving in this PR, having it not open the popover again when clicking on the button while it's already in an open state.

**Comment by peppy** [comment]
I dunno, I didn't mind it reopening each click personally.

**Comment by frenzibyte** [comment]
Hmm, to me it's an established behaviour that clicking on anything that opens something should close it if it's already open (e.g. dropdowns, toolbar toggle buttons). Keeping the behaviour as-is introduces an inconsistency against the "mods" button, since that also acts like a toggle (closes when clicking if open...at least that's how it's supposed to behave).

**Comment by peppy** [comment]
It's probably fine to change it. All the places in operating systems which had behaviour similar to how this used to behave (prior to this PR) have disappeared over the last decade so it does seem like the correct direction.

**Comment by Joehuu** [comment]
> Well, yeah, that's what I'm achieving in this PR, having it not open the popover again when clicking on the button while it's already in an open state.

What I was trying to convey is other popovers opened with buttons should also be changed to this new behavior for consistency.

> keeping the behaviour as-is will already introduce an inconsistency against the "mods" button, since that also acts like a toggle (closes when clicking if open...or at least that's how it's supposed to behave

As you said here, the same applies to the add preset button and presets:

https://github.com/ppy/osu/assets/35318437/0f20da05-af2f-412e-9db6-ff38dbbeaed0

**Comment by frenzibyte** [comment]
I see. For other cases facing the same issue, I would open an issue and solve them separately, as long as we agree that the behaviour proposed by this PR is more correct.

File: osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs [hunk 1]
```diff
@@ -188,9 +188,9 @@ protected override bool OnKeyDown(KeyDownEvent e)
         protected override void UpdateState(ValueChangedEvent<Visibility> state)
         {
             base.UpdateState(state);
-
-            if (state.NewValue == Visibility.Hidden)
-                footerButton.IsActive.Value = false;
+            // intentionally scheduling to let the button have a chance whether the popover will hide from clicking the button or clicking outside
```
Comment (peppy) [inline osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs:191]: This schedule might be okay, but this comment isn't. I read it five times and still can't understand what it's trying to say.

Comment (frenzibyte) [reply osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs:191]: Hmm, it does seem like I've condensed too much context inside. I'll rewrite it.

Comment (frenzibyte) [reply osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs:191]: How does this read?
```diff
diff --git a/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs b/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs
index 648f536bb1..ae928965a3 100644
--- a/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs
+++ b/osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs
@@ -188,8 +188,9 @@ protected override bool OnKeyDown(KeyDownEvent e)
         protected override void UpdateState(ValueChangedEvent<Visibility> state)
         {
             base.UpdateState(state);
-            // intentionally scheduling to let the button have a chance whether the popover will hide from clicking the button or clicking outside
-            // see the "hidingFromClick" field in FooterButtonOptionsV2.
+
+            // a user might be clicking on the footer button itself to close this popover, but the popover responds to the "mouse down" first before the footer button does, due to its order in the input queue.
+            // to give the footer button a chance to know it's being clicked to close the popover (as it relies on overlay state), schedule updating the overlay state to the next frame.
             Schedule(() => footerButton.OverlayState.Value = state.NewValue);
         }
     }
```

Comment (peppy) [reply osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs:191]: I dunno. Based on Joehuu finding at least one other case of this, maybe we should just leave it for now? I don't see it harming anyone (it's not an "annoying" interaction to me)? Doesn't feel important to fix it in this one location any more than others.

Here's another for good measure:


https://github.com/ppy/osu/assets/191335/63d90ce0-4f0a-4484-897d-fffa2981dfc0


I'd say revert and have this PR just fix the visuals.

Comment (frenzibyte) [reply osu.Game/Screens/Select/FooterV2/BeatmapOptionsPopover.cs:191]: Unfortunate, but no problem.

**Review by peppy — APPROVED** [review]

**Pull request merged**